cmd/fpga_plugin/README.md

```diff
@@ -1,120 +1,290 @@
+
+## Dependencies
+
+All components have the same basic dependancies as the
+[generic plugin framework dependencies](../../README.md#about)
+
+To obtain a fully operational FPGA enabled cluster, you must install all three
+major components:
+
+-   [FPGA device plugin](README.md) (this component)
 -   [FPGA admission controller webhook](../fpga_admissionwebhook/README.md)
 -   [FPGA prestart CRI-O hook](../fpga_crihook/README.md)
 
+The CRI-O hook is only *required* if `orchestrated` FPGA bitstream programming mode is
+being used, but is installed by default by the
+[FPGA plugin DaemonSet YAML](../../deployments/fpga_plugin/fpga_plugin.yaml), and is benign
+in `preprogrammed` mode.
 
-### Get source code:
-```
-    $ mkdir -p $GOPATH/src/github.com/intel/
-    $ cd $GOPATH/src/github.com/intel/
-    $ git clone https://github.com/intel/intel-device-plugins-for-kubernetes.git
-```
+If using the `preprogrammed` mode, and therefore *not* using the
+CRI-O prestart hook, runtimes other than CRI-O can be used (that is, the CRI-O hook presently
+*only* works with the CRI-O runtime).
 
-### Build FPGA device plugin:
-```
-    $ cd $GOPATH/src/github.com/intel/intel-device-plugins-for-kubernetes
-    $ make fpga_plugin
-```
+The FPGA device plugin requires a Linux Kernel FPGA driver to be installed and enabled to
+operate. The plugin supports the use of either of two drivers:
 
-### Verify kubelet socket exists in /var/lib/kubelet/device-plugins/ directory:
+- The Linux Kernel in-tree [DFL](https://www.kernel.org/doc/html/latest/fpga/dfl.html) driver
+- The out of tree [OPAE](https://opae.github.io/latest/docs/drv_arch/drv_arch.html) driver
+
+Install this component (FPGA device plugin) first, and then follow the links
+and instructions to install the other components.
+
+## Getting the source code
+
+```bash
+$ go get -d -u github.com/intel/intel-device-plugins-for-kubernetes
 ```
-    $ ls /var/lib/kubelet/device-plugins/kubelet.sock
-    /var/lib/kubelet/device-plugins/kubelet.sock
+
+## Verify node kubelet config
+
+Every node that will be running the FPGA plugin must have the
+[kubelet device-plugins](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/)
+configured. For each node, check that the kubelet device plugin socket exists:
+
+```bash
+$ ls /var/lib/kubelet/device-plugins/kubelet.sock
+/var/lib/kubelet/device-plugins/kubelet.sock
 ```
 
-### Choose mode for FPGA device plugin
+## Deploying as a DaemonSet
 
-You can run the FPGA device plugin in either `af` or `region` mode.
+> **Note:** The DaemonSet YAML launches `af` mode by default.
 
-In `af` mode, the device plugin exposes accelerator functions
-programmed onto the device as resources. Accelerator functions identified
-by the same AFU ID are translated to resources of the same type.
+To deploy the FPGA plugin as a DaemonSet, you first need to build a container image
+for the plugin and ensure that is visible to your nodes.
 
```

Comment:
Building images makes sense only for debugging purposes. Default deployment is configured to use images from the docker hub.

Would it make sense to remove everything related to rebuilding from this document to make deployment instructions short and clear?

Reply:
We could move them to a 'developer' section later in the doc maybe. Would that be acceptable?

Reply:
Sure it's acceptable. Alternatively we can move them into a separate .md file.

Reply:
I've shuffled it to the end of the deployment section and reworded the text before it etc. hopefully that has pushed it far enough down...